httomo/wrappers_class.py

```diff
@@ -0,0 +1,170 @@
+                 module_name: str = None,
+                 function_name: str = None,
+                 method_name: str = None):
+        self.function_name = function_name
+        
+    def _execute_generic(self,
+                         method_name: str,
+                         params: Dict,
+                         data: xp.ndarray) -> xp.ndarray:
+        """The generic wrapper to execute functions for external packages.
+
+        Args:
+            method_name (str): The name of the method to use
+            params (Dict): A dict containing independent of httomo params        
+            data (xp.ndarray): a numpy or cupy data array
+
+        Returns:
+            xp.ndarray: A numpy or cupy array containing processed data.           
+        """
+        # gets module executed
+        data = getattr(self.module, method_name)(data, **params)
+        return data
+    
+    def _execute_normalize(self,
+                          method_name: str,
+                          params: Dict,
+                          data: xp.ndarray,
+                          flats: xp.ndarray,
+                          darks: xp.ndarray) -> xp.ndarray:
+        """For normalisation function we also require flats and darks.
+
+        Args:
+            method_name (str): The name of the method to use
+            params (Dict): A dict containing independent of httomo params
+            data (xp.ndarray): a numpy or cupy data array
+            flats (xp.ndarray): a numpy or cupy flats array
+            darks (xp.ndarray): a numpy or darks flats array
+
+        Returns:
+            xp.ndarray: a numpy or cupy array of normalised data
+        """
+        data = getattr(self.module, method_name)(data, flats, darks, **params)
+        return data
+
+    def _execute_reconstruction(self, 
+                               method_name: str,
+                               params: Dict,
+                               data: xp.ndarray,
+                               angles_radians: np.ndarray) -> xp.ndarray:
+        """The reconstruction wrapper.
+
+        Args:
+            method_name (str): The name of the method to use
+            params (Dict): A dict containing independent of httomo params
+            data (xp.ndarray): a numpy or cupy data array
+            angles_radians (np.ndarray): a numpy array of projection angles
+
+        Returns:
+            xp.ndarray: a numpy or cupy array of the reconstructed data
+        """
+        # for 360 degrees data the angular dimension will be truncated while angles are not.
+        # Truncating angles if the angular dimension has got a different size
+        angular_dim_size = np.size(data, 0)
+        if np.size(data, 0) != len(angles_radians):
+            angles_radians = angles_radians[0:angular_dim_size]
+                
+        data = getattr(self.module, method_name)(data, angles_radians, **params)
+        return data   
+    
+class tomopy_wrapper(wrappers):
```

Comment:
class names are CamelCase: https://google.github.io/styleguide/pyguide.html#3164-guidelines-derived-from-guidos-recommendations
```suggestion
class TomoPyWrapper(wrappers):
```